tests/functional/prometheous/test_prometheus_validations.py

```diff
@@ -974,3 +975,81 @@ def test_restart_prometheus_glusterfs_pod(self):
+                " pod is not present")
+
+        self.prefix = "autotest-{}".format(utils.get_random_str())
+        if not self.registry_sc:
+            self.skipTest(
+                prometheus_pods, "Skipping test as registry "
+                " storage details are not provided")
+        self._registry_project = self.registry_sc.get(
+            'restsecretnamespace')
+
+        # Get one of the prometheus pod name and respective pvc name
+        prometheus_pod = list(prometheus_pods.keys())[0]
+        pvc_custom = ":.spec.volumes[*].persistentVolumeClaim.claimName"
+        pvc_name = openshift_ops.oc_get_custom_resource(
+            self._master, "pod", pvc_custom, prometheus_pod)
+        self.assertTrue(
+            pvc_name, "Failed to get PVC name for prometheus"
+            " pod {}".format(prometheus_pod))
+        pvc_name = pvc_name[0]
+        self.verify_iscsi_sessions_and_multipath(
+            pvc_name, prometheus_pod, rtype='pod',
+            heketi_server_url=self._registry_heketi_server_url,
+            is_registry_gluster=True)
+
+        # Try to fetch metric from the prometheus pod
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_persistentvolumeclaim_info')
+
+        # Create storage class
+        openshift_ops.switch_oc_project(
+            self._master, self._registry_project)
+        self.sc_name = self.create_storage_class(
+            vol_name_prefix=self.prefix, glusterfs_registry=True)
+        self.addCleanup(openshift_ops.switch_oc_project,
+                        self._master, self._registry_project)
+
+        # Create PVCs and app pods
+        pvc_size, pvc_count, batch_count = 1, 5, 5
+        for _ in range(batch_count):
+            test_pvc_names = self.create_and_wait_for_pvcs(
+                pvc_size, pvc_name_prefix=self.prefix,
+                pvc_amount=pvc_count, sc_name=self.sc_name, timeout=600,
+                wait_step=10)
+            self.create_dcs_with_pvc(
+                test_pvc_names, timeout=600, wait_step=5,
+                dc_name_prefix="autotests-dc-with-app-io",
+                space_to_use=1048576)
+
+        # Check from the prometheus pod for the PVC space usage
+        openshift_ops.switch_oc_project(
+            self._master, self._prometheus_project_name)
+        mount_path = "/prometheus"
+        cmd = "oc exec {0} -- df -PT {1} | grep {1}".format(
+            prometheus_pod, mount_path)
+        out = self.cmd_run(cmd)
+        self.assertTrue(out, "Failed to get info about mounted volume. "
+                             "Output is empty.")
+
+        # Try to fetch metric from prometheus pod
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_persistentvolumeclaim_info')
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_pod_spec_volumes_persistentvolumeclaims_info')
+        self.addCleanup(openshift_ops.switch_oc_project,
```

Comment:
Plz have a check before line number 1040 so that we are currently at i.e which project and use that in the cleanup.

Reply:
Can you please refer to this comment https://github.com/gluster/glusterfs-containers-tests/pull/17#discussion_r595902137 and let me know if we still need to add a check before line 1040 like you suggested?

Reply:
Yes, that okay
For cleanup you need to get back to the project from where you were i.e before switching as in 1040, so in clean you need to move there

Reply:
After discussing with @vamahaja ; as issues were seen in TC clean-up steps (if switch to 'current_project' was done), decided to use 'self._registry_project' itself in the addCleanup() at Line 1055